### ISSUE TITLE
[a11y]: DataTable [expansion - default] - heading not in hierarchical order

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14267]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
Following Headings:
DataTable Heading Level 4
Expandable row content Heading Level 6

Heading levels are not in hierarchical order

Heading Level is skipped. H6  is given after H4 which is not recommended.

### Recommended Fix
Heading Level's should be in logical order, Starting from H1, H2, H3, H4, H5 and H6 respectively. It is not recommended to skip the heading levels.

### Reproduction/example
Record `Number 70` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by wkeese** [comment]
Is this ticket just about the DataTable stories?   The only hardcoded heading I see in DataTable is in TableContainer itself, which we filed as #18835.

**Comment by heloiselui** [comment]
Checked with our accessibility team, and this issue still exists. After H2, H6 is used directly.
